src/components/calcite-ui-icon/calcite-ui-icon.tsx

```diff
@@ -0,0 +1,172 @@
+import { Component, h, Element, Prop, State, Watch, Host } from "@stencil/core";
+import { CSS } from "./resources";
+import { getElementDir } from "../../utils/dom";
+import { fetchIcon, scaleToPx } from "./utils";
+import { Scale } from "../../interfaces/UIIcon";
+import { Theme } from "../../interfaces/common";
+
+@Component({
+  assetsDirs: ["assets"],
+  tag: "calcite-ui-icon",
+  styleUrl: "calcite-ui-icon.scss",
+  shadow: true
+})
+export class CalciteUIIcon {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element()
+  el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, the icon will be filled.
+   */
+  @Prop({
+    reflect: true,
+    mutable: true
+  })
+  filled: boolean = false;
+
+  /**
+   * The UI icon to display. Name must match icon from https://esri.github.io/calcite-ui-icons/.
+   */
+  @Prop({
+    reflect: true,
+    mutable: true
+  })
+  icon: string = null;
```

Comment:
Could we get a typing from this file? `https://github.com/Esri/calcite-ui-icons/blob/master/index.d.ts` Maybe we can get them to generate a type for us to use?

Reply:
I generate everything over there, so could do whatever we needed.

Reply:
@paulcpederson Generating a type with all camel/kebab-cased icon names would be ideal. Although, I'm not sure if it that'll be useful outside of this use case.